### DISCUSSION TITLE
Assign user roles at registration with OpenID

### DISCUSSION THREAD

**Discussion by kbychkov** [directus/directus#14941] [Feature Requests]
I've configured the KeyCloak OpenID provider and can authenticate users. But I still need to assign roles manually upon the user registration. It would be very helpful to obtain roles from the `realm_access.roles` attribute of the JWT token and assign it to users automatically just as it works with LDAP groups.

**Comment by jesusmpc** [comment]
Would be great to set the default user role based on the payload of the jwt token when login in from an external provider.

Azure AD for example let's you define app roles over SPN so you can receive a role array as claim on the jwt, then based on that claim define the role of the user when logging in for the first time.

Maybe the approach here could be a new event hook?, maybe a filter one?, something like `auth.define.role` passing the token claims called somewhere like [here](https://github.com/directus/directus/blob/main/api/src/auth/drivers/openid.ts#L187) on the openid provider, this way any developer can implement their own logic to define a custom role based on the JWT payload.

How does it sound?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
